bundles/action/org.openhab.action.pushsafer/src/main/java/org/openhab/action/pushsafer/internal/Pushsafer.java

```diff
@@ -0,0 +1,174 @@
+import java.net.URLEncoder;
+import java.util.Arrays;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.openhab.core.scriptengine.action.ActionDoc;
+import org.openhab.core.scriptengine.action.ParamDoc;
+import org.openhab.io.net.http.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class contains the methods that are made available in scripts and rules
+ * for sending messages via the Pushsafer mobile device push service..
+ *
+ * @author Chris Graham / Kevin Siml
+ * @since 1.5.0
+ */
+public class Pushsafer {
+
+    private static final Logger logger = LoggerFactory.getLogger(Pushsafer.class);
+
+    private final static String API_URL = "https://www.pushsafer.com/api";
+    private final static String CONTENT_TYPE = "application/x-www-form-urlencoded";
+    private final static String UTF_8_ENCODING = "UTF-8";
+
+    private final static String API_RETURN_ROOT_TAG = "hash";
+    private final static String API_RETURN_STATUS_TAG = "status";
+    private final static String API_RETURN_ERROR_TAG = "error";
+    private final static String API_RETURN_STATUS_SUCCESS = ":1,";
+    private final static int API_MAX_MESSAGE_LENGTH = 4096;
+
+    public static final String MESSAGE_KEY_API_KEY = "k";
+    public static final String MESSAGE_KEY_MESSAGE = "m";
+    public static final String MESSAGE_KEY_DEVICE = "d";
+    public static final String MESSAGE_KEY_TITLE = "t";
+    public static final String MESSAGE_KEY_ICON = "i";
+    public static final String MESSAGE_KEY_VIBRATION = "v";
+    public static final String MESSAGE_KEY_SOUND = "s";
+
+    static int timeout = 10000;
+
+    // Primary method for sending a message to the Pushsafer API
+    @
+    ActionDoc(text = "Send a notification to your iOS, Android or Win10 device. Private or Alias Key and message are required. All other can effectively be null. Check the Pushsafer.com API for more informations: https://www.pushsafer.com/en/pushapi", returns = "<code>true</code>, if successful and <code>false</code> otherwise.")
+    public static boolean pushsafer(@ParamDoc(name = "apiKey", text = "Your Private or Alias Key.") String apiKey,
+
+        @ParamDoc(name = "message", text = "Your message.") String message, @ParamDoc(name = "title", text = "Your message's title.") String title, @ParamDoc(name = "device", text = " Your device or device group id to send the message directly to that device(s), rather than all of your devices.") String device, @ParamDoc(name = "icon", text = "A icon to show with your message.") String icon, @ParamDoc(name = "vibration", text = "A title for your supplementary URL, otherwise just the URL is shown.") String vibration, @ParamDoc(name = "sound", text = "The id of one of the sounds to override the default sound.") String sound) {
```

Comment:
"directly to that device(s), rather than all of your devices"

I think this should say "directly to that device, rather than all of your devices", but I find the entire sentence confusing, so I'm not sure. Can you clarify the purpose?